### PR TITLE
[JENKINS-60725] Prevent false-positive warnings about missing classes on Java 11 triggered by JavaMelody when Monitoring Plugin is installed

### PR DESCRIPTION
**Pull request by MRamonLeon**
See [JENKINS-60725](https://issues.jenkins-ci.org/browse/JENKINS-60725).
It is a false positive about a missing java dependency on Java 11. The monitoring plugin calls Java Melody - FilterContext object which throws a CNFE but it's caught and ignored, so we have to avoid warning about that.

Pretty straightforward change, I don't add any test because it may imply slowing down the build, adding dependencies to Java Melody or adding the monitoring plugin with very little value.

### Proposed changelog entries

* Prevent false-positive warnings about missing classes on Java 11 triggered by JavaMelody when Monitoring Plugin is installed

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@jenkinsci/core-pr-reviewers 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by fcojfernandez — APPROVED** [review]

File: core/src/main/java/jenkins/telemetry/impl/java11/MissingClassTelemetry.java [hunk 1]
```diff
@@ -104,7 +104,10 @@
             {"java.util.ResourceBundle$Control", "newBundle"},
             //hundreds when a job is created
             {"org.codehaus.groovy.control.ClassNodeResolver", "tryAsLoaderClassOrScript"},
-            {"org.kohsuke.stapler.RequestImpl$TypePair", "convertJSON"}
+            {"org.kohsuke.stapler.RequestImpl$TypePair", "convertJSON"},
+            //JENKINS-60725
+            {"net.bull.javamelody.FilterContext", "isMojarraAvailable"}
```
Comment (alecharp) [inline core/src/main/java/jenkins/telemetry/impl/java11/MissingClassTelemetry.java]:
```suggestion
            {"net.bull.javamelody.FilterContext", "isMojarraAvailable"} // JENKINS-60725
```

File: core/src/main/java/jenkins/telemetry/impl/java11/MissingClassTelemetry.java [hunk 2]
```diff
@@ -104,7 +104,10 @@
             {"java.util.ResourceBundle$Control", "newBundle"},
             //hundreds when a job is created
             {"org.codehaus.groovy.control.ClassNodeResolver", "tryAsLoaderClassOrScript"},
-            {"org.kohsuke.stapler.RequestImpl$TypePair", "convertJSON"}
+            {"org.kohsuke.stapler.RequestImpl$TypePair", "convertJSON"},
+            //JENKINS-60725
```
Comment (alecharp) [inline core/src/main/java/jenkins/telemetry/impl/java11/MissingClassTelemetry.java]:
```suggestion
```

**Review by alecharp — APPROVED** [review]
Having the comment about the bun on the same line could help future readers, but it's just me being picky. It's fine as it is as well.

**Review by res0nance — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
I plan to merge it tomorrow if no negative feedback and once CI restores.
@MRamonLeon could you please copy-edit the changelog to make it more usser-friendly? E.g. it would be fine to just say "Prevent false-positive warnings about missing JavaMelody classes when Monitoring Plugin is installed" or so

**Comment by MRamonLeon** [comment]
Unrelated errors, closing and reopening

**Comment by timja** [comment]
This is now ready-for-merge

**Comment by oleg-nenashev** [comment]
Yes, now it can be merged. @MRamonLeon do you plan to integrate @alecharp 's suggestion?

**Comment by res0nance** [comment]
Retriggered to get a passing CI build

**Pull request merged**